Cargo.toml

```diff
@@ -10,10 +10,15 @@ categories = ["cryptography", "no-std"]
 keywords = ["miden", "crypto", "hash", "merkle"]
 edition = "2021"
 
+[features]
+alloc = []
+default = ["std", "winter_crypto/default", "winter_math/default", "winter_utils/default"]
+std = ["alloc", "winter_crypto/std", "winter_math/std", "winter_utils/std"]
```

Comment:
Could we avoid defining `alloc` as a feature here? In other crates, we define just `default` and `std` - e.g., something like this:
```
[features]
default = ["std"]
std = ["winter_crypto/std", "winter_math/std", "winter_utils/std"]
```

And then for feature flags we use `#[cfg(not(feature = "std"))]` for `no-std` parts.

Reply:
Agreed! The reason to do that is to sometimes have the types at least represented in super constrained environments that will not have an allocator - such as some hardware wallets.

But until we actually have such applicable need, it might be overkill to activate this split